lib/resources/nginx_conf.rb

```diff
@@ -37,6 +38,21 @@ def params
       @params = {}
     end
 
+    def method_missing(name)
+      # ensure params are loaded
+      @params ||= parse_nginx(@conf_path)
```

Comment:
We should only load data into `@params` in one place... whether that's in `parse_nginx` or another method doesn't matter really. But we need to make it a single place.